### PR TITLE
M3 add new cache bundle 2

### PR DESCRIPTION
**Pull request by hluchas**
Original and closed PR is here https://github.com/mautic/mautic/pull/8817

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | feature
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | /
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/179
| Issue(s) addressed                     | 

#### Description:

This bundle enables the developer to use common Symfony caching mechanism with tag-aware adapters. It implements FS, Memcache and Redis adapters.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Set up an adapter as described in documentation
3. Use the cache as described in documentation
4. Verify items are created (e.g. redis records)
5. Verify items expire
6. Clear cache and verify items are purged

### CODE REVIEW

**Comment by hluchas** [comment]
> Hey @hluchas,
> Something went wrong with the build.
> 
> TravisCI finished with status `errored`, which means the build failed because of something unrelated to the tests, such as a problem with a dependency or the build process itself.
> 
> [View build log](https://travis-ci.org/mautic/mautic/builds/719223969)
> 
> ###### TravisBuddy Request Identifier: 54915730-e200-11ea-81c3-ef1e647c3fbd

~~In this case it is not defined in `.env` (`DB_NAME`) neither in `local.php` (`db_name`) This is problem of missing CI config or we could create workaround.~~

https://github.com/mautic/mautic/blob/23f9ab331545e78aa4a031630cfff16488ce96a1/app/bundles/CacheBundle/Config/config.php#L74

fixed https://github.com/mautic/mautic/pull/9107/commits/94f14fcc9d4d229e4f0dca2b19aebcf4f8d6fc69

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by hluchas** [comment]
Hi @RCheesley !
CLA log https://s3.amazonaws.com/cla-bot-logs/mautic-c892fd28-b45b-47e1-b3c3-fb046112dccd says that
```
2020-09-21T09:36:44.742Z INFO The contributors @galvani have not signed the CLA, adding error status to the pull request
```

Galvani is original author of this PR and I added some corrections. He's not anymore involved in Mautic development. 

What we can do with this?

**Comment by hluchas** [comment]
@RCheesley Could you please take a look here? This is the same prob https://github.com/mautic/mautic/pull/8769#issuecomment-717121945

**Comment by RCheesley** [comment]
All you have to do is call the bot using @cla-bot check ;)

**Comment by mabumusa1** [comment]
@hluchas this is a great feature, I hope we can get it pushed to the next release. I will test it this weekend.

May you please resolve `composer.json` conflict.

**Comment by RCheesley** [comment]
@hluchas looks like a small conflict here to resolve

**Comment by RCheesley** [comment]
We will need to push the documentation from the readme over to https://github.com/mautic/developer-documentation on merge - pinging @mautic/education-team & @mautic/product-team

**Comment by hluchas** [comment]
```
License "GPL-3.0" is a deprecated SPDX license identifier, use "GPL-3.0-only" or "GPL-3.0-or-later" instead
require.doctrine/orm : exact version constraints (2.7.0) should be avoided if the package follows semantic versioning
require.mrclay/minify : exact version constraints (2.2.0) should be avoided if the package follows semantic versioning
Error: Process completed with exit code 2.
```

It's working on my local with PHP7.3 and Composer `1.10.17 `

**Comment by hluchas** [comment]
~~This will be easy to fix. We are using Composer version 1.10.17. 
@fedys Am I right?~~

EDIT: Correct version of Composer is used

**Comment by dennisameling** [comment]
@hluchas the actual problem is this (see CI logs), you can ignore the `composer validate` warnings:

> The lock file is not up to date with the latest changes in composer.json, it is recommended that you run `composer update` or `composer update <package name>`.

Could you please try `composer update predis/predis` locally? That should update the `composer.lock` file with the new hash which you can then commit. Thanks!!

**Comment by fedys** [comment]
@hluchas @dennisameling FYI, if you need to update just the hash you can run `composer update --lock`.

**Comment by hluchas** [comment]
> @hluchas @dennisameling FYI, if you need to update just the hash you can run `composer update --lock`.

This is resolved ;-)

File: app/bundles/CacheBundle/README.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+# mautic-cache-plugin
+
+Enables PSR-6 and PSR-16 caching. Check: [Symfony Cache Component](https://symfony.com/doc/3.4/components/cache.html)
+
+## Installation
+
+Install this plugin as submodule into plugins directory.
```
Comment (escopecz) [inline app/bundles/CacheBundle/README.md]: @hluchas please remove this readme. I've moved the docs to https://github.com/mautic/developer-documentation/pull/179 and updated them. The installation steps are not up to date at the moment. It'll be easier to keep the docs up to date in one place only.

**Review by escopecz — DISMISSED** [review]
We use this bundle heavily in our fork and this PR is blocking a lot of others that build on top of this. The code actually does not do anything to existing functionality so it cannot break anything.

👍

File: app/bundles/CacheBundle/README.md (hunk 1 above)
Comment (hluchas) [reply app/bundles/CacheBundle/README.md]: removed ...

**Review by escopecz — DISMISSED** [review]

**Review by escopecz — DISMISSED** [review]

**Review by escopecz — APPROVED** [review]

**Pull request merged**